apps/namadillo/src/hooks/useExtensionEvents.tsx

```diff
@@ -0,0 +1,23 @@
+import { useEventListenerOnce } from "@namada/hooks";
+import { Events } from "@namada/types";
+import { useAtomValue, useSetAtom } from "jotai";
+import { accountBalanceAtom, defaultAccountAtom } from "slices/accounts";
+import { namadaExtensionConnectionStatus } from "slices/settings";
+
+export const useExtensionEvents = (): void => {
+  const defaultAccount = useAtomValue(defaultAccountAtom);
+  const balances = useAtomValue(accountBalanceAtom);
+  const setNamadaExtensionConnected = useSetAtom(
+    namadaExtensionConnectionStatus
+  );
+
+  // Register handlers:
+  useEventListenerOnce(Events.AccountChanged, async () => {
+    await defaultAccount.refetch();
+    balances.refetch();
+  });
+
+  useEventListenerOnce(Events.ConnectionRevoked, () =>
+    setNamadaExtensionConnected("idle")
```

Comment:
Not introduced in this PR, but `ConnectionRevoked` gets fired when any connection is revoked, so interfaces listening to this event have to check whether they are still connected in the event handler. If I'm remembering correctly, it's this way because we couldn't know which tab the extension should send the event to, so it sends it to all tabs.

Reply:
Can we address this one in a new PR?

Reply:
Added as #898!